sanitizers/template.go

```diff
@@ -0,0 +1,27 @@
+package sanitizers
+
+import (
+	htmlTemplate "html/template"
+	"strings"
+	textTemplate "text/template"
+)
+
+const evilTemplateCommand = "{{ .EvilCommand }}"
+
+func ParseHTMLTemplate(fakePC int, tmpl *htmlTemplate.Template, text string) (*htmlTemplate.Template, error) {
+	if strings.Contains(text, evilTemplateCommand) {
+		ReportFinding("Template Injection")
+	} else {
+		GuideTowardsContainment(text, evilTemplateCommand, fakePC)
+	}
+	return tmpl.Parse(text)
```

Comment:
I wonder if `Parse` is the correct method to hook. It shouldn't be an issue to create a "problematic" template, as long as it's not executed with `Execute`.

Reply:
IMO `Parse` is a valid hooking point to just check whether injection is possible at all. That said, I agree that we should hook `Execute` and `ExecuteTemplate` as well to find those cases where the tampered with template is actually "*used*".

Reply:
Good point! Parsing the `Parse` function may lead to false positives. I've changed the logic to hook `Execute` and `ExecuteTemplate` and check whether the template to be executed contains the injected action.